src/app/home/problem-list/problem-list.component.ts

```diff
@@ -21,4 +26,12 @@ export class ProblemListComponent implements OnInit {
       }); // TODO: Handle error
     });
   }
+
+  onClick(problem: IProblem): void {
+    problem.serverId = 'sqlInjection'; // FIXME Get this from firebase. Or find a better way.
+    this.router.navigate(['home/tutorial'], {
+      // TODO What all do we need? And does it make sense as a query parameter? Side effect: refreshing the page will have different behaviours in each case
+      state: { problem }
```

Comment:
I am not sure if we need this `state` at this time. Also are we using this in Tutorials page?

Reply:
Yes. This is how we are transferring the `serverId` of the problem from the problems page to the tutorials page. My larger idea is to maybe move this to query parameters so that even if the user refreshes the page, we have the details in the URL. It is one of the most common techniques to store user's current state in the url itself.

Reply:
Yeah I read the comments that we have not yet decided. Query params is one way to do it. We can think about it.